### PR TITLE
Different default directories for CLI and ASB

### LINKED ISSUE #437: Different default `data-dir` for ASB and CLI

### PR DESCRIPTION
**Pull request by da-kami**
Fixes #437 

Using the same default directory as data-/config-dir has caused unwanted side effects when running both applications on the same machine.
Use these directory names:
- ASB: `xmr-btc-swap-asb`
- CLI: `xmr-btc-swap-cli`

Since the functionality is now application specific the respective functions were moved into the appropriate module of the application.

### CODE REVIEW

**Review by bonomat — COMMENTED** [review]
Nice. One thing to fix please, the other thing is just an idea.

File: swap/src/asb/config.rs [hunk 1]
```diff
@@ -99,6 +100,24 @@ pub fn read_config(config_path: PathBuf) -> Result<Result<Config, ConfigNotIniti
     Ok(Ok(file))
 }
 
+/// This is to store the configuration and seed files
+// Linux: /home/<user>/.config/xmr-btc-swap/config.toml
+// OSX: /Users/<user>/Library/Preferences/xmr-btc-swap/config.toml
+pub fn default_config_path() -> Result<PathBuf> {
+    ProjectDirs::from("", "", "xmr-btc-swap-asb")
```
[the remark this answers is not in the document]
Comment (da-kami) [reply swap/src/asb/config.rs:107]: Thanks! Forgot to update that comment 😬

File: swap/src/asb/config.rs [hunk 2]
```diff
@@ -99,6 +100,24 @@ pub fn read_config(config_path: PathBuf) -> Result<Result<Config, ConfigNotIniti
...
+        .map(|proj_dirs| proj_dirs.config_dir().to_path_buf())
+        .map(|dir| Path::join(&dir, "config.toml"))
+        .context("Could not generate default configuration path")
+}
+
+/// This is the default location for storing data for the ASB
+// Linux: /home/<user>/.local/share/xmr-btc-swap-asb/
+// OSX: /Users/<user>/Library/Application Support/xmr-btc-swap-asb/
+fn asb_default_data_dir() -> Option<std::path::PathBuf> {
+    ProjectDirs::from("", "", "xmr-btc-swap-asb")
```
[the remark this answers is not in the document]
Comment (da-kami) [reply swap/src/asb/config.rs:117]: I do like the idea - not sure about the `network` though, I find that a bit to high-level.
Initially I thought about going for `xmr-btc-swap/cli` and `xmr-btc-swap/asb` but I was not sure what is better.

How about:

`ProjectDirs::from("", "xmr-btc-swap", "asb")`
`ProjectDirs::from("", "xmr-btc-swap", "cli")`

Or:

`ProjectDirs::from("comit", "xmr-btc-swap", "asb")`
`ProjectDirs::from("comit", "xmr-btc-swap", "cli")`

@thomaseizinger @rishflab what are your thoughts on that?

[the remark this answers is not in the document]
Comment (da-kami) [reply swap/src/asb/config.rs:117]: Reading the docs of `ProjectDirs::from` I see why you specified it as `ProjectDirs::from("network", "comit", "xmr-btc-swap-asb")` - maybe our qualifier could be `xmr-btc` though?

```
ProjectDirs::from("xmr-btc", "comit", "asb")
ProjectDirs::from("xmr-btc", "comit", "cli")
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -7,6 +7,15 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
 ## [Unreleased]
 
+### Fixed
+
+- An issue where both the ASB and the CLI point to the same default directory `xmr-btc-swap` for storing data.
+  The asb now uses `xmr-btc-swap-asb` and the CLI `xmr-btc-swap-cli` as default directory.
+  This is a breaking change.
+  If you want to access data created by a previous version you will have to rename the data folder or one of the following.
+  For the CLI you can use `--data-dir` to point to the old directory.
+  For the ASB you can change the data-dir in the config file of the ASB.
```
Comment (thomaseizinger) [inline CHANGELOG.md:17]: You probably wanted this to be an enumeration I guess? (Untested what this renders to)

```suggestion
  If you want to access data created by a previous version you will have to rename the data folder or one of the following:
  1. For the CLI you can use `--data-dir` to point to the old directory.
  2. For the ASB you can change the data-dir in the config file of the ASB.
```

File: swap/src/asb/config.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (thomaseizinger) [reply swap/src/asb/config.rs:117]: Yeah this is a bit of a mess. You can put the specifier there but it is not gonna have any effect on Linux. The application name will still need to be globally unique. That is why we didn't bother with it in the past.

**Review by thomaseizinger — APPROVED** [review]
Geez this reminds me how much the names in this project suck :sweat_smile: 

Fixes the problem though so ACK.

File: swap/src/asb/config.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (bonomat) [reply swap/src/asb/config.rs:117]: One thing you could do is create a sub-folder in whatever you received back from `ProjectDirs::from("", "", "xmr-btc-swap")`. That way we have a sub-folder for `asb` and `cli` on all platforms. 

This is not a critical point for me, so please ignore and move on if you don't feel like it :)

**Review by bonomat — APPROVED** [review]

File: CHANGELOG.md (hunk 3 above)
Comment (da-kami) [reply CHANGELOG.md]: `dprint` not complaining locally - CLion renders it like this:

image

**Comment by da-kami** [comment]
@bonomat @thomaseizinger I gave it another shot going with a `system` wide folder and then application folders (`asb`, `cli`) on top of that:

```
xmr-btc-swap/asb
xmr-btc-swap/cli
```

in the end it won't make too much difference I guess, but I agree with @bonomat that it would be nice if both applications are bundled in the same folder.

File: swap/src/asb/config.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (da-kami) [reply swap/src/asb/config.rs]: done :)

**Review by bonomat — APPROVED** [review]
Cool 👍

**Pull request merged**